test/passing/break_cases.ml

```diff
@@ -0,0 +1,28 @@
+let f x = function
+  | C
+  | P
+  | A
+  | K -> 1
```

Comment:
Here, imagine that the patterns above were long and complex, with nesting, parens, etc. In such cases spotting the `-> 1` can be very hard, leading to misreading the code thinking that the body for `D` also applies to e.g. `A`. I don't know how to avoid this danger without breaking after `->` or changing the indentation of `|` for the nested cases.

Reply:
I don't think that's a problem in practice (at least I haven't met that case yet). It's rare to have long complex patterns and or clause (I think?). 

Also to be fair, I would tend to indent this function as:

```ocaml
let f x = function
  | A | B | C -> foo
  | D             -> bar
```

But I am not sure an automated tool can handle this case properly... (and I am to change my "code style" to something more standard)

Reply:
It could very well be a rare occurrence, but we have hit it enough that I would avoid using such a mode on our code.

Modulo alignment, it should be possible to get your formatting by force-breaking between the cases, but not between the nested or-patterns. I think it would make sense to force-break both the cases and or-patterns with `--sparse --break-cases`, and to force-break the cases but not the nested or-patterns with just `--break-cases`.